src/iop/grain.c

```diff
@@ -327,6 +335,58 @@ static double _simplex_2d_noise(double x, double y, uint32_t octaves, double per
   return total;
 }
 
+float paper_resp(float exposure, float mb, float gp)
+{
+  float density;
+  float delta = GRAIN_LUT_DELTA_MAX * exp(mb * log(GRAIN_LUT_DELTA_MIN));
+  density = (1 + 2 * delta) / (1 + exp( (4 * gp * (0.5 - exposure)) / (1 + 2 * delta) )) - delta;
```

Comment:
All these suffix-less constants seem scary, too easy to unintentionally do int-based operation i'd say.
Append `.0f`